lib/blacklight/marc/indexer.rb

```diff
@@ -0,0 +1,54 @@
+require 'library_stdnums'
+ 
+require 'traject/macros/marc21_semantics'
+ 
+require 'traject/macros/marc_format_classifier'
+ 
+require 'traject/solr_json_writer'
+require 'traject/marc_reader'
+class Blacklight::Marc::Indexer < Traject::Indexer
+  autoload :Dewey, 'blacklight/marc/indexer/dewey'
+  autoload :Formats, 'blacklight/marc/indexer/formats'
+
+  include  Traject::Macros::Marc21Semantics
+  include Traject::Macros::MarcFormats
+
+  ATOZ = ('a'..'z').to_a.join('')
+  ATOU = ('a'..'u').to_a.join('')
+
+  def initialize
+    super
+
+    settings do
+      provide "reader_class_name", "Traject::MarcReader"
+      provide "marc_source.type", "binary"
```

Comment:
as far as I know, the ruby-marc reader dispatches to an appropriate parser. But I'll look for some xml to test against.